log/zap.go

```diff
@@ -198,6 +198,14 @@ func ZContext(ctx context.Context) zap.Field {
 	return zap.Inline(&marshalledContext{Context: ctx})
 }
 
+func NiceZapError(err error) zap.Field {
+	var loggable ObjectMarshaller
+	if errors.As(err, &loggable) {
+		return zap.Inline(loggable)
+	}
```

Comment:
With `errors.Join()` you can unwrap the individual errors with

```go
	if errBatch, ok := err.(interface{ Unwrap() []error }); ok {
		return zap.Errors(errBatch.Unwrap())
	}
```

The cast to an unnamed interface is a bit ugly but `errors.Unwrap()` cannot handle joined errors unfortunately 😕